webui/react/src/pages/WorkspaceDetails.tsx

```diff
@@ -219,6 +219,7 @@ const WorkspaceDetails: React.FC = () => {
   ]);
 
   const fetchAll = useCallback(async () => {
+    if (!canViewWorkspace({ workspace: { id } })) return;
```

Comment:
i think `canViewWorkspace({ workspace: { id } })` should be outside of `usecallback` to prevent infinite api calls.

Reply:
Without this line, `fetchRolesAssignableToScope` later on was being called infinitely and it errors if you don't have access to the workspace.

Reply:
I agree it would be cleaner to write differently but how can it disable fetchAll from outside of the callback?

Reply:
i mean this

```diff
diff --git a/webui/react/src/pages/WorkspaceDetails.tsx b/webui/react/src/pages/WorkspaceDetails.tsx
index 12882b554..a5a30f9af 100644
--- a/webui/react/src/pages/WorkspaceDetails.tsx
+++ b/webui/react/src/pages/WorkspaceDetails.tsx
@@ -218,8 +218,9 @@ const WorkspaceDetails: React.FC = () => {
     workspaceAssignments,
   ]);
 
+  const canViewWorkspaceFlag = canViewWorkspace({ workspace: { id } });
   const fetchAll = useCallback(async () => {
-    if (!canViewWorkspace({ workspace: { id } })) return;
+    if (!canViewWorkspaceFlag) return;
     await Promise.allSettled([
       fetchWorkspace(),
       fetchUsers(),
@@ -228,11 +229,10 @@ const WorkspaceDetails: React.FC = () => {
       fetchRolesAssignableToScope(),
     ]);
   }, [
-    id,
-    canViewWorkspace,
+    canViewWorkspaceFlag,
     fetchWorkspace,
-    fetchGroups,
     fetchUsers,
+    fetchGroups,
     fetchGroupsAndUsersAssignedToWorkspace,
     fetchRolesAssignableToScope,
   ]);
```

Reply:
oh i see what you meant, but there's infinite loop with the current approach.

https://user-images.githubusercontent.com/109113616/217928246-d0e77c73-e849-4f4d-8cd7-a586d9d8026b.mov